File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        ArrayList<Integer>[] bucket = (ArrayList<Integer>[]) new ArrayList[nums.length];
        Arrays.sort(nums);
        for (int i = 0; i < bucket.length; i++) {
            bucket[i] = new ArrayList<Integer>();
        }
        int currentNum = nums[0];
        int index = 1;
        for(int i = 1;i<nums.length;i++){
            if(currentNum == nums[i]){
                index++;
            }
            else{
                bucket[index-1].add(currentNum);
                index = 1;
                currentNum = nums[i];
            }
        }
        System.out.println(Arrays.toString(bucket));
        bucket[index-1].add(currentNum);

        int[] answer = new int[k];
        index = k-1;
        System.out.println(Arrays.toString(bucket));
        for(int i = bucket.length-1;i>=0;i--){
            for(int j: bucket[i]){
                if(index == -1){
                    return answer;
                }
                answer[index] = j;
                index--;
            }
        }
        return answer;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-2.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n == 1){
            return 1;
        }
        if(n == 2){
            return 2;
        }
        std::vector<int> dp(n+1, 0);
        dp[n] = 1;
        dp[n-1] = 1;
        for(int i = n-2;i>=0;--i){
            dp[i] = dp[i+1]+dp[i+2];
        }
        return dp[0];

    }
};
